contracts/SSVNetwork.sol

```diff
@@ -100,56 +157,121 @@ contract SSVNetwork is OwnableUpgradeable, ISSVNetwork {
+            _updateOperatorFeeUnsafe(operatorId, operatorFee);
+        } else {
+        */
+        _operatorFeeChangeRequests[operatorId] = OperatorFeeChangeRequest(
+            fee.shrink(),
+            uint64(block.timestamp) + _declareOperatorFeePeriod,
+            uint64(block.timestamp) + _declareOperatorFeePeriod + _executeOperatorFeePeriod
+        );
+        emit OperatorFeeDeclaration(msg.sender, operatorId, block.number, fee.shrinkable());
+    }
+
+    function cancelDeclaredOperatorFee(uint64 operatorId) onlyOperatorOwnerOrContractOwner(operatorId) external {
+        OperatorFeeChangeRequest memory feeChangeRequest = _operatorFeeChangeRequests[operatorId];
+
+        if(feeChangeRequest.fee == 0) {
+            revert NoPendingFeeChangeRequest();
+        }
+
+        delete _operatorFeeChangeRequests[operatorId];
+
+        emit DeclaredOperatorFeeCancelation(msg.sender, operatorId);
+    }
+
+    function executeOperatorFee(uint64 operatorId) onlyOperatorOwnerOrContractOwner(operatorId) external {
+        OperatorFeeChangeRequest memory feeChangeRequest = _operatorFeeChangeRequests[operatorId];
+
+        if(feeChangeRequest.fee == 0) {
+            revert NoPendingFeeChangeRequest();
+        }
+
+        if(block.timestamp < feeChangeRequest.approvalBeginTime || block.timestamp > feeChangeRequest.approvalEndTime) {
+            revert ApprovalNotWithinTimeframe();
+        }
+
+        _updateOperatorFeeUnsafe(operatorId, feeChangeRequest.fee);
+
+        delete _operatorFeeChangeRequests[operatorId];
+    }
+
+    function getOperatorDeclaredFee(uint64 operatorId) external view returns (uint256, uint256, uint256) {
+        OperatorFeeChangeRequest memory feeChangeRequest = _operatorFeeChangeRequests[operatorId];
+
+        if(feeChangeRequest.fee == 0) {
+            revert NoPendingFeeChangeRequest();
+        }
+
+        return (feeChangeRequest.fee.expand(), feeChangeRequest.approvalBeginTime, feeChangeRequest.approvalEndTime);
+    }
+
+    function getOperatorFee(uint64 operatorId) external view returns (uint256) {
+        Operator memory operator = _operators[operatorId];
+
+        if (operator.owner == address(0)) revert OperatorNotFound();
 
-        emit OperatorFeeSet(operatorId, fee);
+        return operator.fee.expand();
     }
 
     function registerValidator(
         bytes calldata publicKey,
-        uint64[] memory operatorIds,
-        bytes calldata shares,
-        uint64 amount
+        bytes32 clusterId,
+        bytes calldata shares
     ) external {
-        _validateValidatorParams(
-            operatorIds,
-            publicKey
-        );
+        if (publicKey.length != 48) {
+            revert InvalidPublicKeyLength();
+        }
+
+        uint64[] memory operatorIds = _clusters[clusterId].operatorIds;
 
         // Operator[] memory operators;
-        bytes32 clusterId = _getOrCreateCluster(operatorIds);
         bytes32 hashedValidator = keccak256(publicKey);
```

Comment:
should be moved here, no?
if (_validatorPKs[hashedValidator].clusterId > 0) {
            revert ValidatorAlreadyExists();
        }